### ISSUE TITLE
[tx] segfault in test suite

### ISSUE THREAD

**Issue by bnavigator** [adobe-type-tools/afdko#1480]
I try to update the afdko package for openSUSE. The test suite fails with a segfault when it calls tx.

- Build log with all package versions and CMake configurations: [afdko-buildlog-py39.txt](https://github.com/adobe-type-tools/afdko/files/8239058/afdko-buildlog-py39.txt)
- Test log: [afdko-buildlog-py39-tests.txt](https://github.com/adobe-type-tools/afdko/files/8239059/afdko-buildlog-py39-tests.txt)

Backtrace:

```
abuild@voyagerS9:~/rpmbuild/BUILD/afdko-3.8.1> gdb tx    
GNU gdb (GDB; openSUSE Tumbleweed) 11.1
Copyright (C) 2021 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-suse-linux".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://bugs.opensuse.org/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from tx...
Reading symbols from /usr/lib/debug/usr/bin/afdko-tx-3.9-3.8.1-0.x86_64.debug...
(gdb) run -t1 /home/abuild/rpmbuild/BUILD/afdko-3.8.1/tests/tx_data/input/cid_roundtrip/testCID.ufo subset.ufo
Starting program: /usr/bin/tx -t1 /home/abuild/rpmbuild/BUILD/afdko-3.8.1/tests/tx_data/input/cid_roundtrip/testCID.ufo subset.ufo
Download failed: No route to host.  Continuing without debug info for /lib64/ld-linux-x86-64.so.2.
Download failed: No route to host.  Continuing without debug info for /home/abuild/rpmbuild/BUILD/afdko-3.8.1/system-supplied DSO at 0x7ffff7fc1000.
Missing separate debuginfos, use: zypper install glibc-debuginfo-2.35-2.1.x86_64
Download failed: No route to host.  Continuing without debug info for /lib64/libm.so.6.
Download failed: No route to host.  Continuing without debug info for /lib64/libc.so.6.
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".

Program received signal SIGSEGV, Segmentation fault.
writeRealArrayDef (h=h@entry=0x555555636a80, key=key@entry=0x55555560a515 "OtherBlues", cnt=-4991471925828486353, array=0x7ffff7eb6000, array@entry=0x7ffff7a261d8) at /usr/src/debug/python-afdko-3.8.1-0.x86_64/c/shared/source/t1write/t1write.c:763
763             float value = *array++;
(gdb) bt
#0  writeRealArrayDef (h=h@entry=0x555555636a80, key=key@entry=0x55555560a515 "OtherBlues", cnt=-4991471925828486353, array=0x7ffff7eb6000, array@entry=0x7ffff7a261d8)
    at /usr/src/debug/python-afdko-3.8.1-0.x86_64/c/shared/source/t1write/t1write.c:763
#1  0x00005555555bddc6 in writeRealArrayDef (array=0x7ffff7a261d8, cnt=<optimized out>, key=0x55555560a515 "OtherBlues", h=0x555555636a80)
    at /usr/src/debug/python-afdko-3.8.1-0.x86_64/c/shared/source/t1write/t1write.c:1097
#2  writePrivateDict (h=h@entry=0x555555636a80, private=private@entry=0x7ffff7a26048, SDBytes=SDBytes@entry=1)
    at /usr/src/debug/python-afdko-3.8.1-0.x86_64/c/shared/source/t1write/t1write.c:1096
#3  0x00005555555bf1e0 in writeFDArray (SDBytes=1, FDMap=0x7fffffffe2b0 "\001\001", FDCnt=<optimized out>, h=0x555555636a80)
    at /usr/src/debug/python-afdko-3.8.1-0.x86_64/c/shared/source/t1write/t1write.c:1535
#4  writeHostCIDKeyedFont (h=h@entry=0x555555636a80) at /usr/src/debug/python-afdko-3.8.1-0.x86_64/c/shared/source/t1write/t1write.c:1862
#5  0x00005555555c05d5 in writeCIDKeyedFont (h=0x555555636a80) at /usr/src/debug/python-afdko-3.8.1-0.x86_64/c/shared/source/t1write/t1write.c:1914
#6  t1wEndFont (h=0x555555636a80, top=0x555555647980) at /usr/src/debug/python-afdko-3.8.1-0.x86_64/c/shared/source/t1write/t1write.c:2162
#7  0x000055555557d39b in t1_EndFont (h=0x55555562f2a0) at /usr/src/debug/python-afdko-3.8.1-0.x86_64/c/shared/source/tx_shared/tx_shared.c:2547
#8  0x0000555555581495 in ufoReadFont (h=h@entry=0x55555562f2a0, origin=<optimized out>)
    at /usr/src/debug/python-afdko-3.8.1-0.x86_64/c/shared/source/tx_shared/tx_shared.c:4688
#9  0x000055555557417c in doFile (h=h@entry=0x55555562f2a0, 
    srcname=srcname@entry=0x7fffffffebb6 "/home/abuild/rpmbuild/BUILD/afdko-3.8.1/tests/tx_data/input/cid_roundtrip/testCID.ufo")
    at /usr/src/debug/python-afdko-3.8.1-0.x86_64/c/tx/source/tx.c:452
#10 0x000055555557436e in doSingleFileSet (h=0x55555562f2a0, srcname=0x7fffffffebb6 "/home/abuild/rpmbuild/BUILD/afdko-3.8.1/tests/tx_data/input/cid_roundtrip/testCID.ufo")
    at /usr/src/debug/python-afdko-3.8.1-0.x86_64/c/tx/source/tx.c:502
#11 0x00005555555752df in parseArgs (h=0x55555562f2a0, argc=3, argv=0x7fffffffe8a0) at /usr/src/debug/python-afdko-3.8.1-0.x86_64/c/tx/source/tx.c:572
#12 0x0000555555573981 in main (argc=3, argv=0x7fffffffe8a0) at /usr/src/debug/python-afdko-3.8.1-0.x86_64/c/tx/source/tx.c:1594
(gdb) 
```

```
abuild@voyagerS9:~/rpmbuild/BUILD/afdko-3.8.1> tx -v
Versions:
    tx        1.2.6
    absfont   1.0.54
    ctutil    2.0.3
    dynarr    2.0.4
    sfntread  1.0.7
    t2cstr    1.0.22
    cffread   2.1.3
    cffwrite  1.0.56
    sfntwrite 1.0.6
    pstoken   2.0.11
    t1cstr    1.0.21
    t1read    1.0.45
    ttread    1.0.22
    cfembed   2.0.25
    pdfwrite  1.0.7
    svread    1.0.8
    t1write   1.0.35
    svwrite   1.1.11
    uforead   1.1.2
    ufowrite  1.0.9
    varread   1.0.8
```

**Comment by kaydeearts** [comment]
Hi @bnavigator , thank you for filing this issue. I believe this issue should be fixed in the [newest release v3.8.2](https://github.com/adobe-type-tools/afdko/releases/tag/3.8.2). Please let me know if the issue is or isn't resolved — I'll further look into it if not. Thank you!

**Comment by bnavigator** [comment]
Fixed by #1497. Thanks.
